### PR TITLE
[#167347769] Update approach to modal on FiscalCode screen

### PR DESCRIPTION
**Pull request by valentinacala**
This pr introduces the usage of light modals to render the fiscal code in landscape mode. 

Shifting to this approach, they are no more available:
- animation scolling on the bask side of the fac-simile while the preview of the back side is dispayed
- animation when the moal is displayed (not available on the light modal itself)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#167347769](https://www.pivotaltracker.com/story/show/167347769): L'unico item ad utilizzare il compoentne <Modal> deve essere IdentificationModal

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/profile/FiscalCodeScreen.tsx [hunk 1]
```diff
@@ -60,6 +62,18 @@ class FiscalCodeScreen extends React.PureComponent<Props, State> {
       showBackSide: false
```
Comment (fpersico) [inline ts/screens/profile/FiscalCodeScreen.tsx]: State seems no more used.

Comment (valentinacala) [reply ts/screens/profile/FiscalCodeScreen.tsx]: Removed within https://github.com/teamdigitale/io-app/pull/1228/commits/06be2928ad563ae9d4c778207e20722200040403

File: ts/components/FiscalCodeLandscapeOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import FiscalCodeComponent from "./FiscalCodeComponent";
+import AppHeader from "./ui/AppHeader";
+
+type Props = Readonly<{
+  onCancel: () => void;
+  profile: UserProfile;
+  showBackSide?: boolean;
+}>;
+
+const globalHeaderHeight: number =
```
Comment (Undermaken) [inline ts/components/FiscalCodeLandscapeOverlay.tsx]: you could use _Platform.select_
```suggestion
const globalHeaderHeight: number = Platform.select({
  ios: isIphoneX()
    ? customVariables.appHeaderHeight + 42
    : customVariables.appHeaderHeight + 18,
  android: customVariables.appHeaderHeight
});
```

File: ts/components/FiscalCodeLandscapeOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    position: "absolute",
+    right: customVariables.contentPadding,
+    top: globalHeaderHeight - customVariables.appHeaderHeight
+  }
+});
+
+export default class FiscalCodeLandscapeOverlay extends React.PureComponent<
+  Props
+> {
+  // private contentRef: any;
```
Comment (Undermaken) [inline ts/components/FiscalCodeLandscapeOverlay.tsx]: remove commented code

File: ts/components/FiscalCodeLandscapeOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        <AppHeader noLeft={true} dark={true}>
+          <Body />
+        </AppHeader>
+        <StatusBar
+          backgroundColor={customVariables.brandDarkGray}
+          barStyle={"light-content"}
+        />
+        <Content
+          style={styles.content}
+          // tslint:disable-next-line no-object-mutation
```
Comment (Undermaken) [inline ts/components/FiscalCodeLandscapeOverlay.tsx]: remove commented code

File: ts/screens/profile/FiscalCodeScreen.tsx [hunk 5]
```diff
@@ -52,14 +49,18 @@ const styles = StyleSheet.create({
...
-class FiscalCodeScreen extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      showAsLandscape: false,
-      showBackSide: false
-    };
-  }
+class FiscalCodeScreen extends React.PureComponent<Props> {
+  private showModal = () => {
```
Comment (Undermaken) [inline ts/screens/profile/FiscalCodeScreen.tsx]: instead using _// tslint:disable-next-line:no-unused-expression_ could this code be written as follow?
```suggestion
 private showModal = () => {
    if (this.props.profile !== undefined) {
      this.props.showModal(
        <FiscalCodeLandscapeOverlay
          onCancel={this.props.hideModal}
          profile={this.props.profile}
        />
      );
    }
  };
```

Comment (valentinacala) [reply ts/screens/profile/FiscalCodeScreen.tsx]: updated within https://github.com/teamdigitale/io-app/pull/1228/commits/0dc2755797e6953ba75d1ed3fe8604c3ef5ba966

File: ts/components/FiscalCodeLandscapeOverlay.tsx (hunk 4 above)
Comment (valentinacala) [reply ts/components/FiscalCodeLandscapeOverlay.tsx]: updated within https://github.com/teamdigitale/io-app/commit/0dc2755797e6953ba75d1ed3fe8604c3ef5ba966

File: ts/components/FiscalCodeLandscapeOverlay.tsx (hunk 3 above)
Comment (valentinacala) [reply ts/components/FiscalCodeLandscapeOverlay.tsx]: updated within https://github.com/teamdigitale/io-app/commit/0dc2755797e6953ba75d1ed3fe8604c3ef5ba966

File: ts/components/FiscalCodeLandscapeOverlay.tsx (hunk 2 above)
Comment (valentinacala) [reply ts/components/FiscalCodeLandscapeOverlay.tsx]: updated within https://github.com/teamdigitale/io-app/pull/1228/commits/6f9d3e6aed837b3bd9a901114de05fbebbfd19eb

File: ts/components/FiscalCodeLandscapeOverlay.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import AppHeader from "./ui/AppHeader";
+
+type Props = Readonly<{
+  onCancel: () => void;
+  profile: UserProfile;
+  showBackSide?: boolean;
+}>;
+
+const globalHeaderHeight: number = Platform.select({
+  ios: isIphoneX()
```
Comment (Undermaken) [inline ts/components/FiscalCodeLandscapeOverlay.tsx]: Just realized we can do it better, sorry ;)
```suggestion
  ios: customVariables.appHeaderHeight + (isIphoneX() ? 42 : 18),
```

Comment (valentinacala) [reply ts/components/FiscalCodeLandscapeOverlay.tsx]: right, updated!

**Comment by Undermaken** [comment]
@fpersico Could you check if all requested changes are addressed?

**Pull request merged**